ui/components/ui/truncated-definition-list/truncated-definition-list.js

```diff
@@ -18,7 +18,39 @@ export default function TruncatedDefinitionList({
   const [isPopoverOpen, setIsPopoverOpen] = useState(false);
   const t = useI18nContext();
 
-  return (
+  return process.env.CHAIN_PERMISSIONS ? (
```

Comment:
Rather than branching the entire render, could we instead only branch when the output changes?

The top level `Box` for example is identical as are other nested elements.

Reply:
Created a separate function to handle this and avoid repetition here https://github.com/MetaMask/metamask-extension/pull/25777/commits/e87feb11f778de6e03ad95819414cbb2233c4b80